Sm 146 pagination

sanketbambalkar1996 opened the pull request.

santoshgaikwad14 commented on skill-matrix/src/main/java/com/clairvoyant/services/skillmatrix/controller/UserController.java:
```diff
@@ -21,15 +21,26 @@
     }
 
+    @GetMapping("/{page}/{size}")
+    public List<UserResponseDto> getAllUsersWithPagination(@RequestParam(name = "isActive",defaultValue = "true") Optional<Boolean> isActive,
```
- RequestParam extract values from the query string, we don't required  `/{page}/{size}` on GetMapping, it will generate uri something like  `/user?isActive=true&page=1&size=10`
- Are we providing a `isActive` option in user search API ? How we are handling that usecase?

santoshgaikwad14 commented on skill-matrix/src/main/java/com/clairvoyant/services/skillmatrix/controller/UserController.java:
```diff
@@ -39,9 +50,8 @@ public UserResponseDto getUserById(@PathVariable String userId,@RequestParam(nam
     public Status addOrUpdateUser(@RequestBody UserDto userDto) {
 
         LOGGER.info(userDto);
```
Remove this unwanted  LOGGER else make it meaningful.

santoshgaikwad14 commented on skill-matrix/src/main/java/com/clairvoyant/services/skillmatrix/controller/UserController.java:
```diff
@@ -21,15 +21,26 @@
+    @GetMapping("/{page}/{size}")
+    public List<UserResponseDto> getAllUsersWithPagination(@RequestParam(name = "isActive",defaultValue = "true") Optional<Boolean> isActive,
+                                             @PathVariable int page,@PathVariable int size ) {
+        return userService.findAll(isActive, page, size);
```
How are we managing the total items, total page or current page details ? Does UI not require these details for your next page call?

santoshgaikwad14 commented on skill-matrix/src/main/java/com/clairvoyant/services/skillmatrix/service/impl/UserServiceImpl.java:
```diff
@@ -52,29 +54,21 @@ public UserServiceImpl(UserRepository userRepository, UserDesignationService use
+                result.get().setPassword(PasswordUtil.encode(userDto.getPassword()));
+                result.get().setGrade(userDto.getGrade());
+
                 user = result.get();
```
Change [this](https://github.com/teamclairvoyant/skills-base/pull/52/files#diff-c0e9ef055e44d58b6c64d0f3f169c7064ee21fa28de50ae4e3da5b5d767267a6R61-R67) code with something like this.

Use user = result.get();
user.setName(userDto.getName());
...
...
 userRepository.save(user);

santoshgaikwad14 commented on skill-matrix/src/main/java/com/clairvoyant/services/skillmatrix/service/impl/UserServiceImpl.java:
```diff
@@ -131,24 +125,35 @@ public UserResponseDto findById(String id, Optional<Boolean> isActive) {
+                roles.add(role);
+            }
+        } else {
+            throw new ResourceNotFoundException("Roles is not available for user : " + result.get().getId());
```
How this code work for user registration / add user. While creating a new user there will no role mapping present for new user ?

santoshgaikwad14 commented:
@sanketbambalkar1996 Did we addressed above review comments? Does this PR still valid ? as we have new PR https://github.com/teamclairvoyant/skills-base/pull/56 with new changes.

sanketbambalkar1996 commented:
> @sanketbambalkar1996 Did we addressed above review comments? Does this PR still valid ? as we have new PR #56 with new changes.

@santoshgaikwad14  As discussed with Akash . He mentioned me to write searching and pagination in single api . Hence I have raised PR for new Story . I have resolved most of the above comments in new PR .

santoshgaikwad14 commented:
As discussed with @sanketbambalkar1996 , We don't required this change , we created new [PR](https://github.com/teamclairvoyant/skills-base/pull/56/files) having a latest changes.
